wgpu-hal/src/dx12/mod.rs

```diff
@@ -1485,3 +1490,21 @@ impl crate::Queue for Queue {
         (1_000_000_000.0 / frequency as f64) as f32
     }
 }
+#[derive(Debug)]
+pub struct DxilPassthroughShader {
+    pub shader: Vec<u8>,
+    pub entry_point: String,
+    pub num_workgroups: (u32, u32, u32),
+}
+#[derive(Debug)]
+pub struct HlslPassthroughShader {
+    pub shader: String,
+    pub entry_point: String,
+    pub num_workgroups: (u32, u32, u32),
+}
+#[derive(Debug)]
+pub enum ShaderModuleSource {
+    Naga(crate::NagaShader),
+    DxilPassthrough(DxilPassthroughShader),
+    HlslPassthrough(HlslPassthroughShader),
+}
```

Comment:
```suggestion
#[derive(Debug)]
pub struct DxilPassthroughShader {
    pub shader: Vec<u8>,
    pub entry_point: String,
    pub num_workgroups: (u32, u32, u32),
}

#[derive(Debug)]
pub struct HlslPassthroughShader {
    pub shader: String,
    pub entry_point: String,
    pub num_workgroups: (u32, u32, u32),
}

#[derive(Debug)]
pub enum ShaderModuleSource {
    Naga(crate::NagaShader),
    DxilPassthrough(DxilPassthroughShader),
    HlslPassthrough(HlslPassthroughShader),
}
```

Let the structs breath 😆